### ISSUE TITLE
HttpEnvironmentProxy fails when proxy URI contains trailing slash

### ISSUE THREAD

**Issue by markekraus** [dotnet/corefx#32635]
On linux, if the `http_proxy` or `https_proxy` environment variables contain a trailing slash, the proxy will not be used.

This can be reproed with PowerShell 6.1.0:

```powershell
$assembly = [System.Net.Http.HttpClientHandler].Assembly
$class = $assembly.GetType('System.Net.Http.HttpEnvironmentProxy')
$method = $class.GetMethod('TryCreate')
$proxy = [System.Net.WebProxy]::new()
$env:HTTPS_PROXY = 'http://webproxy:3128/'
$method.Invoke($null, @($proxy))
```

it should be `True`, instead it is `False`. 

However, if the trailing slash is removed, TryCreate() works:

```powershell
$assembly = [System.Net.Http.HttpClientHandler].Assembly
$class = $assembly.GetType('System.Net.Http.HttpEnvironmentProxy')
$method = $class.GetMethod('TryCreate')
$proxy = [System.Net.WebProxy]::new()
# No trailing slash
$env:HTTPS_PROXY = 'http://webproxy:3128'
$method.Invoke($null, @($proxy))
```

The issue appears to be in `GetUriFromString`.

```powershell
$assembly = [System.Net.Http.HttpClientHandler].Assembly
$class = $assembly.GetType('System.Net.Http.HttpEnvironmentProxy')
$Method = $class.GetMethod('GetUriFromString',[System.Reflection.BindingFlags]'NonPublic, Static')
```

```powershell
$Method.Invoke($null, @('http://webproxy:3128'))
```

Returns a uri

```powershell
$Method.Invoke($null, @('http://webproxy:3128/'))
```

returns null.

**Comment by stephentoub** [comment]
This looks like a dup of https://github.com/dotnet/corefx/issues/31326, which was fixed by https://github.com/dotnet/corefx/pull/31232 and https://github.com/dotnet/corefx/pull/31327.

**Comment by iSazonov** [comment]
@stephentoub Is the fix in .Net Core 2.1.5?

**Comment by stephentoub** [comment]
I believe so.  What version are you using?

**Comment by iSazonov** [comment]
We moved to .Net Core 2.1.5 in PowerShell master branch but @markekraus repo script doesn't work for me and I can not confirm that fix works.

**Comment by iSazonov** [comment]
I found that the fix is in 2.1.5 https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1.5/2.1.5-commits.md#corefx

**Comment by davidsh** [comment]
Duplicate of #31326.
